capsules/src/public_key_crypto/rsa_keys.rs

```diff
@@ -0,0 +1,197 @@
+//! Helper library for RSA public and private keys
+
+use core::convert::TryInto;
+use core::ops::Range;
+use kernel::hil::entropy;
+use kernel::hil::entropy::{Client8, Continue};
+use kernel::hil::public_key_crypto::keys::{
+    PubKey, PubPrivKey, PubPrivKeyGenerate, PubPrivKeyGenerateClient, RsaKey, RsaPrivKey,
+};
+use kernel::utilities::cells::OptionalCell;
+use kernel::utilities::mut_imut_buffer::MutImutBuffer;
+use kernel::ErrorCode;
+
+/// A Public/Private RSA 2048 key pair
+pub struct RSA2048Keys<'a> {
+    public_key: OptionalCell<MutImutBuffer<'static, u8>>,
+    private_key: OptionalCell<MutImutBuffer<'static, u8>>,
+    client: OptionalCell<&'a dyn PubPrivKeyGenerateClient<'a>>,
+}
+
+impl<'a> Client8 for RSA2048Keys<'a> {
+    fn entropy_available(
+        &self,
+        entropy: &mut dyn Iterator<Item = u8>,
+        error: Result<(), ErrorCode>,
+    ) -> Continue {
+        error.unwrap();
+
+        let pub_key = self.public_key.take().unwrap();
+
+        match pub_key {
+            MutImutBuffer::Mutable(buf) => {
+                for (i, b) in entropy.enumerate() {
```

Comment:
Should this be generating and storing a set of primes?

Reply:
Yeah, I have a partial implementation of generating keys. The prime tests are tricky to implement with an async Rng engine.

Besides that generating keys is complex and I think should be reviewed in a separate PR, with the aim of getting lots of eyes on it.

Right now I would just like to focus on getting the HIL approved and then go from there.